tests/periph_gpio_arduino/main.sketch

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (C) 2018 Tom Keddie
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       Manual test application for GPIO peripheral drivers using
+ *              Arduino pin numbering
+ *
+ * @author      Tom Keddie <github@bronwenandtom.com>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "shell.h"
+#include "periph/gpio.h"
+
+static int get_digital_pin(const char* pinstring, int* pin)
+{
+    int arduino_pin = -1;
+    char *endptr  = NULL;
+
+    if (*pinstring == 'A') {
+        // TODO : add analogInputToDigitalPin macro to arduino_board.h for all boards
+        return -1;
+    }
+
+    arduino_pin = strtol(pinstring, &endptr, 0);
+
+    if (*endptr != '\0' || arduino_pin == -1 || arduino_pin > (int)(sizeof(arduino_pinmap)/sizeof(arduino_pinmap[0]))) {
+        return -1;
+    }
+
+    *pin = arduino_pin;
```

Comment:
why use a parameter and not the function return value, its an `int` anyway, so `< 0` indicates error and `>= 0` is the pin